agent/config/builder.go

```diff
@@ -2522,6 +2523,12 @@ func validateAbsoluteURLPath(p string) error {
 func (b *builder) buildTLSConfig(rt RuntimeConfig, t TLS) (tlsutil.Config, error) {
 	var c tlsutil.Config
 
+	// This flag indicates whether any of the per-listener configuration was set.
+	// The flag was populated earlier when applying deprecated options.
```

Comment:
Could we clarify this comment with whether this is for backwards compatibility and when we can expect to remove it?